pkg/object/meshcontroller/worker/worker.go

```diff
@@ -362,115 +359,45 @@ func (wrk *Worker) updateHearbeat() error {
-			logger.Errorf("add egress scope watching service: %s failed: %v", serviceName, err)
-			return
-		}
-	}
-
-	handleServiceInstances := func(instanceKvs map[string]*spec.ServiceInstanceSpec) (continueWatch bool) {
-		continueWatch = true
-		defer func() {
-			if err := recover(); err != nil {
-				logger.Errorf("%s: recover from: %v, stack trace:\n%s\n",
-					wrk.superSpec.Name(), err, debug.Stack())
-			}
-		}()
-		logger.Infof("handle informer egress service: %s's instance update event, ins: %#v", serviceName, instanceKvs)
-		serviceSpec := wrk.service.GetServiceSpec(serviceName)
-
-		var instanceSpecs []*spec.ServiceInstanceSpec
-		for _, v := range instanceKvs {
-			instanceSpecs = append(instanceSpecs, v)
-		}
-		if err := wrk.egressServer.UpdatePipeline(serviceSpec, instanceSpecs); err != nil {
-			logger.Errorf("handle informer egress failed, update service: %s failed: %v", serviceName, err)
-		}
-
-		return
-	}
-
-	if err := wrk.informer.OnServiceInstanceSpecs(serviceName, handleServiceInstances); err != nil {
-		if err != informer.ErrAlreadyWatched {
-			logger.Errorf("add egress prefix watching service: %s failed: %v", serviceName, err)
-			return
-		}
-	}
+	return worker.store.Put(layout.ServiceInstanceStatusKey(worker.serviceName, worker.instanceID), string(buff))
 }
 
-func (wrk *Worker) informJavaAgent() error {
+func (worker *Worker) informJavaAgent() error {
 	handleServiceSpec := func(event informer.Event, service *spec.Service) bool {
 		switch event.EventType {
 		case informer.EventDelete:
 			return false
 		case informer.EventUpdate:
-			if err := wrk.observabilityManager.UpdateService(service, event.RawKV.Version); err != nil {
+			if err := worker.observabilityManager.UpdateService(service, event.RawKV.Version); err != nil {
 				logger.Errorf("update service %s failed: %v", service.Name, err)
 			}
 		}
 
 		return true
 	}
 
-	err := wrk.informer.OnPartOfServiceSpec(wrk.serviceName, informer.AllParts, handleServiceSpec)
+	err := worker.informer.OnPartOfServiceSpec(worker.serviceName, informer.AllParts, handleServiceSpec)
 	if err != nil && err != informer.ErrAlreadyWatched {
 		return fmt.Errorf("on informer for observability failed: %v", err)
 	}
 
 	return nil
 }
 
-func (wrk *Worker) watchEvent() {
-	for {
-		select {
-		case <-wrk.done:
-			return
-		case name := <-wrk.egressEvent:
-			logger.Infof("add egress wanted watching service: %s", name)
-			wrk.addEgressWatching(name)
-		}
-	}
-}
-
 // Status returns the status of worker.
-func (wrk *Worker) Status() *supervisor.Status {
+func (worker *Worker) Status() *supervisor.Status {
 	return &supervisor.Status{
 		ObjectStatus: nil,
 	}
 }
 
 // Close close the worker
-func (wrk *Worker) Close() {
-	close(wrk.done)
-
-	wrk.informer.Close()
-	wrk.registryServer.Close()
-	wrk.apiServer.Close()
+func (worker *Worker) Close() {
+	close(worker.done)
+
+	worker.egressServer.Close()
+	worker.ingressServer.Close()
+	worker.informer.Close()
```

Comment:
`informer` should be closed first, this is to ensure no notifications could be sent after `egressServer` & `ingressServer` were closed, otherwise, they will be re-initialized.

also need to update informer implementation, to make sure no notification be sent after it is closed.

Reply:
The `Informoer`'s implementation update I recommend to put it into the next PR. 
And let me update the sequence of `close` callings.

Reply:
I've made the changes to `Informer`, but I agree with you to hold it on to the next PR, as it introduces new `wait` operations and may result in deadlocks.
I'll create an issue to track this.